src/openrct2/entity/Yaw.hpp

```diff
@@ -18,28 +18,46 @@ namespace OpenRCT2::Entity::Yaw
     // smallest precision is 4 frames
     constexpr const uint8_t PrecisionOffset[] = { 3, 2, 1, 0, 0, 0, 0, 0 };
 
-    constexpr int32_t YawTo4(int32_t yaw)
+    constexpr [[nodiscard]] int32_t MaskYaw(int32_t yaw)
+    {
+        return yaw & 0x1F;
+    }
+    constexpr [[nodiscard]] uint8_t YawFrom4(int32_t rotation)
+    {
+        return rotation << 3;
+    }
+    constexpr [[nodiscard]] int32_t YawTo4(int32_t yaw)
     {
         return yaw >> 3;
     }
-    constexpr int32_t YawTo8(int32_t yaw)
+    constexpr [[nodiscard]] int32_t YawTo8(int32_t yaw)
     {
         return yaw >> 2;
     }
-    constexpr int32_t YawTo16(int32_t yaw)
+    constexpr [[nodiscard]] int32_t YawTo16(int32_t yaw)
     {
         return yaw >> 1;
     }
-    constexpr int32_t YawTo32(int32_t yaw)
+    constexpr [[nodiscard]] int32_t YawTo32(int32_t yaw)
     {
         return yaw;
     }
-    constexpr int32_t YawToPrecision(int32_t yaw, uint8_t precision)
+    constexpr [[nodiscard]] int32_t YawToPrecision(int32_t yaw, uint8_t precision)
     {
         return yaw >> PrecisionOffset[precision];
     }
-    constexpr uint8_t NumSpritesPrecision(uint8_t precision)
+    constexpr [[nodiscard]] uint8_t NumSpritesPrecision(uint8_t precision)
     {
         return 4 << precision;
     }
+    constexpr uint8_t [[nodiscard]] PrecisionFromNumSprites(uint8_t numSprites)
```

Comment:
What is the planned use for this? Do we really need something like this perhaps a table would be easier for most readers to understand.